File: hw/ramos/u2/hw19Inheritance/Java/Farmsystem/src/main/java/ec/edu/espe/farmsystem/model/Cow.java
package ec.edu.espe.farmsystem.model;

import java.util.Date;

/**
 *
 * @author Paulo Ramos
 */
public class Cow extends FarmAnimal {

    private boolean isProducingMilk;
    private float milkQuantityPerDay;

    public Cow(boolean isProducingMilk, float milkQuantityPerDay, int id, String breed, Date bornOn,
            String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.isProducingMilk = isProducingMilk;
        this.milkQuantityPerDay = milkQuantityPerDay;
    }

    public float milk() {
        if (isProducingMilk) {
            return milkQuantityPerDay;
        } else {
            return 0;
        }
    }

    @Override
    public String toString() {
        return "Cow{"
                + "isProducingMilk=" + isProducingMilk
                + ", milkQuantityPerDay=" + milkQuantityPerDay
                + ", " + super.toString()
                + '}';
    }

    public boolean isIsProducingMilk() {
        return isProducingMilk;
    }

    public void setIsProducingMilk(boolean isProducingMilk) {
        this.isProducingMilk = isProducingMilk;
    }

    public float getMilkQuantityPerDay() {
        return milkQuantityPerDay;
    }

    public void setMilkQuantityPerDay(float milkQuantityPerDay) {
        this.milkQuantityPerDay = milkQuantityPerDay;
    }
}

File: hw/chicaiza/u2/hw19Inheritance/FarmSystemPython/model/cow.py
from model.farm_animal import FarmAnimal

class Cow(FarmAnimal):
    def __init__(self, is_producing_milk: bool, liters_a_day: float, milk: float,
                 id, breed, born_on, gender, is_able_to_reproduce, weight, cage):
        super().__init__(id, breed, born_on, gender, is_able_to_reproduce, weight, cage)
        self.is_producing_milk = is_producing_milk
        self.liters_a_day = liters_a_day
        self.milk = milk

    def __str__(self):
        return f"Cow(producingMilk={self.is_producing_milk}, litersDay={self.liters_a_day}, {super().__str__()})"
